src/app/pages/system/view-enclosure/view-enclosure.component.ts

```diff
@@ -148,6 +149,10 @@ export class ViewEnclosureComponent implements OnDestroy {
       this.system.sensorData = evt.data;
     });
 
+    core.register({ observerClass: this, eventName: 'Resilvering' }).pipe(untilDestroyed(this)).subscribe((evt: ResilverEvent) => {
+      if (evt.data.scan.state == 'FINISHED') this.fetchData();
```

Comment:
It would be better to use an enum here.
Probably, `PoolScanState` is most suitable in this case.
https://github.com/truenas/webui/blob/443a13760275f4f78b07aeedf96baf59459b3690/src/app/enums/pool-scan-state.enum.ts#L1-L4

We also have used `JobState.Finished` for some unknown reason here:
https://github.com/truenas/webui/blob/db68e1b4172d2880dc81769a6f21765503fd70a2/src/app/components/common/topbar/topbar.component.ts#L232-L237

Reply:
Will replace string with enum